internal/server/distribution.go

```diff
@@ -37,8 +37,8 @@ func ReadDistributions(distro string) ([]Distribution, error) {
 				return nil
 			}
 
-			f, err := os.Open(path)
-			defer f.Close()
+			f, err := os.Open(path) // #nosec
+			defer f.Close() // #nosec
```

Comment:
Why does gosec need to be disabled here?

Reply:
It gets upset about the deferred close, but most people seem to disable the gosec checks here. I am not good enough with golang to know if there's a better pattern for this.

Do you have any ideas?

Reply:
At the least, I'd say we need a comment stating which check is hidden here. Do you know if it is "G307: Deferring a method which returns an error"?

At least that looks like the most plausible to me. If it is, I think it's fine to ignore the error returned by `Close`, because we're catching any read errors from the file in `json.Decoder.Decode()`. Deferring in a loop is not the best idea anyway, though -- some linters complain about that already.

Reply:
That's the one. It's `G307`.

Reply:
Good to know.

1. Do you mind adding a comment/link mentioning that – so that the person touching this code next doesn't need to guess. Something like `// Ignore gosec's "G307: Deferring a method which returns an error"` should be enough.

2. Please remove it on the `Open` call. We do check the error there.

Reply:
@larskarlitski I added some comments in there and re-pushed.